File: app/src/main/java/com/hk/ijournal/models/repository/localdata/dao/UserDao.java
package com.hk.ijournal.models.repository.localdata.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.hk.ijournal.models.DiaryUser;

@Dao
public interface UserDao {
    @Insert
    void insertUser(DiaryUser user);

    @Delete
    void deleteUser(DiaryUser user);

    @Query("select * from usertable where username like :userName")
    DiaryUser getUserbyName(String userName);
}

File: app/src/main/java/com/hk/ijournal/repository/data/source/local/dao/UserDao.kt
package com.hk.ijournal.repository.data.source.local.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.hk.ijournal.repository.data.source.local.entities.User

@Dao
interface UserDao {
    @Insert
    suspend fun insertUser(user: User): Long

    @Delete
    suspend fun deleteUser(user: User)

    @Query("select * from usertable where username = :userName")
    suspend fun getUserbyName(userName: String?): User?

    @Query("select * from usertable where uid = :uid")
    suspend fun getUserbyId(uid: Long): User?
}